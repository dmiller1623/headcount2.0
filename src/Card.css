.cards {
  box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2), 0 6px 20px 0 rgba(0, 0, 0, 0.19);
  background-image: linear-gradient(to top, #fff1eb 0%, #ace0f9 100%);  border: solid 2px black;
  border-radius: 15px;
  margin: .2rem;
  opacity: .85;
  width: 15.5%;
}

.cards:hover {
  opacity: 1;
  cursor: pointer;
}

h2 {
  color: #f3f7f8;
  font-family: 'Space Mono', monospace;
  font-size: 1.3rem;
  text-align: center;
  text-shadow: 0px 2px 3px #555;
}

ul {
  font-size: .8rem;
  margin: .1rem .3rem .4rem .3rem;
  text-align: center;
  padding: 0;
}

.highNum {
  color:#021331;
}

.lowNum {
  color: #3bbfc3;
}

.selectedCard {
  background-image: linear-gradient(to top, #fff1eb 0%, #ace0f9 100%);  border: solid 2px black;
  display: flex;
  flex-direction: column;
  border: solid 4px red;
  border-radius: 15px;
  margin: .2rem;
  width: 15.5%;
}