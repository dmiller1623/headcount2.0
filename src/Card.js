import React from 'react'
import './Card.css'


const Card = ({ location, stats }) => {
  console.log(location)
  return (
   <div className="cards">
    <h2>{location}</h2>
      {
        Object.keys(stats).map(stat => {
          return (
            <article>
              <ul>
                {stat}: {stats[stat]}
              </ul>
            </article>
          )
        })
      }
   </div>
  )
}

export default Card