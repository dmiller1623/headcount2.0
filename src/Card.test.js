import React from 'react';
import ReactDOM from 'react-dom';
import Card from './Card';
import { shallow, mount } from 'enzyme';

describe('Card', () => {
  let wrapper
  let displaySelectedMock
  let key 
  let displayComparedMock
  let removeSelectedMock
  let selectedCardsArray = []
  beforeEach(() => {
    key = 1
    displaySelectedMock = jest.fn()
    displayComparedMock = jest.fn()
    removeSelectedMock = jest.fn()
    wrapper = shallow(<Card 
      location= 'COLORADO'
      stats={ {2004: 0.24} }
      key={key}
      selected
      selectedCards={selectedCardsArray}
      displaySelected={displaySelectedMock}
      displayCompared={displayComparedMock}
      removeSelected={removeSelectedMock}
      />)
  })
    it('expect wrapper to match snapshot', () => {
        expect(wrapper).toMatchSnapshot()
    })
    it('handle click updates state', () => {
      expect(wrapper.state('clicked')).toEqual(false)
      wrapper.instance().handleClick()
      expect(wrapper.state('clicked')).toEqual(true)
      
    })
    it('calls removeSelected when a card is clicked on and clicked is true', () => {
      let location = "COLORADO"
      wrapper.instance().handleClick()
      wrapper.setState( { clicked: true } )
      expect(displaySelectedMock).not.toHaveBeenCalled()
      expect(removeSelectedMock).toHaveBeenCalledWith(location)
    })
    it('calls display selelected when clicked is false', () => {
      wrapper = shallow(<Card 
        location= 'COLORADO'
        stats={ {2004: 0.24} }
        key={key}
        selectedCards={selectedCardsArray}
        displaySelected={displaySelectedMock}
        displayCompared={displayComparedMock}
        removeSelected={removeSelectedMock}
        />)
      wrapper.instance().handleClick()
      expect(displaySelectedMock).toHaveBeenCalled()
    })
    it('should have a class of highnum if less than .5', () => {
      expect(wrapper.find('ul').hasClass('highNum')).toEqual(true)
    })
    it('should have a class of lowNum if more than lowNum', () => {
      wrapper = shallow(<Card 
        location= 'COLORADO'
        stats={ {2004: 0.6} }
        key={key}
        selectedCards={selectedCardsArray}
        displaySelected={displaySelectedMock}
        displayCompared={displayComparedMock}
        removeSelected={removeSelectedMock}
        />)
      
      expect(wrapper.find('ul').hasClass('lowNum')).toEqual(true)
    })

})