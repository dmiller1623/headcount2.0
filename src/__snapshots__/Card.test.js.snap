// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card expect wrapper to match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    displayCompared={[Function]}
    displaySelected={[Function]}
    location="COLORADO"
    removeSelected={[Function]}
    selected={true}
    selectedCards={Array []}
    stats={
        Object {
            "2004": 0.24,
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          COLORADO
</h2>,
        Array [
          <article>
            <ul
                        className="highNum"
            >
                        2004
                        : 
                        0.24
            </ul>
</article>,
        ],
      ],
      "className": "selectedCard",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "COLORADO",
        },
        "ref": null,
        "rendered": "COLORADO",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <ul
            className="highNum"
>
            2004
            : 
            0.24
</ul>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "2004",
              ": ",
              0.24,
            ],
            "className": "highNum",
          },
          "ref": null,
          "rendered": Array [
            "2004",
            ": ",
            0.24,
          ],
          "type": "ul",
        },
        "type": "article",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            COLORADO
</h2>,
          Array [
            <article>
              <ul
                            className="highNum"
              >
                            2004
                            : 
                            0.24
              </ul>
</article>,
          ],
        ],
        "className": "selectedCard",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "COLORADO",
          },
          "ref": null,
          "rendered": "COLORADO",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <ul
              className="highNum"
>
              2004
              : 
              0.24
</ul>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "2004",
                ": ",
                0.24,
              ],
              "className": "highNum",
            },
            "ref": null,
            "rendered": Array [
              "2004",
              ": ",
              0.24,
            ],
            "type": "ul",
          },
          "type": "article",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
